### ISSUE TITLE
e.sent.data.forEach is not a function when getting invalid JSON response

### ISSUE THREAD

**Issue by L3P3** [axios/axios#3675]
#### Describe the bug
I am working with a (not really clean/modern) PHP backend I don't have any control over. When there is a server error, it sometimes responds with bare php error printouts or simple html code, not formatted in JSON. In these cases, we don't get something like a "invalid format" exception from axios we could handle, but instead a cryptic JS error saying `e.sent.data.forEach is not a function`.
We had this issue from day one, about 2 years ago. Since it is such an apparently obvious bug, we thought that just waiting will fix it. Nope. Wonder why no one else had this issue. :thinking: 

#### To Reproduce
```js
axios.post("myapi", ["some", "data", "to", "post"])
```
Normally, we get JSON responses like this:
```js
{success: true, message: "Transaction done. Good day!"}
```
Or this:
```js
{success: false, message: "I'm afraid I can't do that, Dave."}
```
Now, consider this myapi responds with something like this:
```html
<h1>Permission denied</h1>
Please log in to see this page!
```
It were quite ugly to handle these cases via a `SyntaxError` and parsing the html alone but it is even uglier to catch a random, undocumented JS error thrown inside axios.

#### Expected behavior
I expect to get a `SyntaxError` or something even more dedicated in case a JSON response is not actually valid JSON.

#### Environment
 - Axios Version 0.19.2
 - Adapter no idea/does not matter
 - Browser any
 - Browser Version any
 - Node.js Version none
 - OS: any
 - Additional Library Versions: none

#### Additional context/Screenshots
I am afraid I cannot give you any more screenshots than the message already stated in a message box.
Also cannot give you a stack trace for now as I currently do not have time for setting up the app locally.

**Comment by DigitalBrainJS** [comment]
`Wonder why no one else had this issue.`
#61, #3419

**Comment by L3P3** [comment]
> `Wonder why no one else had this issue.`
> #61, #3419

The key difference here is that I get an internal JS error that is not documented whatsoever.

**Comment by prashanthgundas** [comment]
i'm facing same issue getting string data from API
![11](https://user-images.githubusercontent.com/44539772/111957653-4ba97100-8b12-11eb-9235-4b57568f2927.png)
 response

**Comment by DigitalBrainJS** [comment]
> e.sent.data.forEach is not a function
> I get an internal JS error 
@L3P3 Hmm, are you sure? Can you share the error stack? The Axios source code doesn't contain the `e.sent.data.forEach` string, moreover, it's written in ES3, so it doesn't use Array.prototype.forEach method at all.
I'm pretty much sure, this is the error that your handler is throwing. And if so, just check the type of the data field before iterating over it as an array. If Axios cannot parse the JSON, it leaves the response as a string and currently, it doesn't throw any SyntaxError exceptions.

**Comment by jasonsaayman** [comment]
Fixed in #3688
